File: Back/LockerZone/LockerZone.Domain/Dtos/Auth/TokenDto.cs

namespace LockerZone.Domain.Dtos
{
    public class TokenDto
    {
        public string Token { get; set; }=string.Empty;
        public DateTime Expiration { get; set; }
        public ApplicationUserDto CurrentUser { get; set; } = default!;
        public string RoleName { get; set; } = string.Empty;
        public bool IsActive { get; set; }
    }

    public class ApplicationUserDto
    {
        public string? UserName { get; set;}
        public string? FullName { get; set; }
        public string RoleName { get; set; } = default!;
        public string? ProfilePicture { get; set; }
    }

}

File: Front/LockerZone/src/app/models/auth/TokenDto.ts
export interface TokenDto {
  token: string;
  expiration: string;
  currentUser: ApplicationUserDto;
  isActive: boolean;
}
export interface ApplicationUserDto {
  userName: string | null;
  fullName: string | null;
  roleName: string;
  profilePicture: string | null;
}
